### PR TITLE
[WFLY-18682] Upgrading WildFly Http Client to 2.0.5.Final

### PR DESCRIPTION
**Pull request by ropalka**
https://issues.redhat.com/browse/WFLY-18682

### CODE REVIEW

File: ee-feature-pack/galleon-shared/src/main/resources/modules/system/layers/base/org/wildfly/http-client/naming/main/module.xml [hunk 1]
```diff
@@ -15,6 +15,7 @@
         <module name="java.naming"/>
         <module name="org.wildfly.http-client.common"/>
         <module name="org.wildfly.client.config"/>
+        <module name="org.jboss.ejb-client" />
```
Comment (darranl) [inline ee-feature-pack/galleon-shared/src/main/resources/modules/system/layers/base/org/wildfly/http-client/naming/main/module.xml:18]: @ropalka does this mean there will also need to be a change to dependencies on clients using WildFly HTTP Client?

Comment (ropalka) [reply ee-feature-pack/galleon-shared/src/main/resources/modules/system/layers/base/org/wildfly/http-client/naming/main/module.xml:18]: No when such clients use it either via maven dependency or JBoss Modules dependency mechanism @darranl .

Comment (darranl) [reply ee-feature-pack/galleon-shared/src/main/resources/modules/system/layers/base/org/wildfly/http-client/naming/main/module.xml:18]: On the clients though are we sure they will have the required dependency, they could use a strategy like we do ourselves and exclude transitive dependencies?

**Comment by darranl** [comment]
Just noticed a couple of errors in the Windows job looked directly related to the HTTP Client, running that job again.

File: ee-feature-pack/galleon-shared/src/main/resources/modules/system/layers/base/org/wildfly/http-client/naming/main/module.xml (hunk 1 above)
Comment (ropalka) [reply ee-feature-pack/galleon-shared/src/main/resources/modules/system/layers/base/org/wildfly/http-client/naming/main/module.xml:18]: If ejb-client will not be on the classpath they will get ClassNotFoundException and will need to include it @darranl

**Comment by ropalka** [comment]
I just upgraded to 2.0.4.Final version that should sort out the Windows issues @darranl

**Comment by darranl** [comment]
@ropalka FYI I am still seeing the same CI failure on the run from today.

**Comment by darranl** [comment]
Thank you very much @ropalka and @tadamski - CI is green so merging.

**Review by darranl — APPROVED** [review]

**Pull request merged**